### DISCUSSION TITLE
restore files

### DISCUSSION THREAD

**Discussion by tjhunkin-inhance** [stefanzweifel/laravel-backup-restore#51] [Ideas]
during the backup, I have some images that are zipped too, would be great to restore them as well

**Comment by stefanzweifel** [comment]
This feature went through my head as well. I didn't implement this as it can become complicated quite quickly.

Some general questions and thoughts on this:

- Where are the image located? In `/public/`? In `/storage/app/`? Is the folder configured as a file-system disk?
- Should the package restore **all** files in the backup? Including `/app/`, `/resource/`?
  - Chances are high that the package could destroy code changes you haven't committed yet.
  - This could potentially also override your local `.env`-file.
- Would you like to be able to select the file-system disk that should be restored? 
- Should the package remove all existing files in the "target location" (eg. your local app) before restoring the files?

**Reply by tjhunkin-inhance** [reply]
I see, makes sense.

**Reply by stefanzweifel** [reply]
@tjhunkin-inhance I don't want to dismiss the feature. A genuine question for your project: Where are the files in your project located and where would they have to be restored too?

**Reply by tjhunkin-inhance** [reply]
@stefanzweifel I use Spatie Media Library Pro, which uploads the images to public\storage locally. I usually wouldn't use local storage, but Media Library Pro for Livewire 3 doesn't allow multiple-image selection to upload to S3 or R2. It might be an option to specify paths in config or a single path as an argument.

**Reply by ojsoft** [reply]
Well, I for example only backup the storage app/public folder, the rest is in gitlab and does not need to be in the backup.

What would really help though and I bet this is easy, if you have an option to keep the extracted backup folder in existence after the backup restore, so one can then copy the necessary. files over by hand. I have like 10 GB Database but 150 GB files. It is stupid to download the zip two times for a complete backup restore process. If I see this right, you delete anything afterwards in the process.

**Reply by stefanzweifel** [reply]
@ojsoft That's a great idea and something I wanted to add in a future release. 

I myself also think it's sometimes quite the hassle to redownload everything again, just to debug the backup. Might even create a secondary command like `backup:download` just for this. 🤔

**Reply by ojsoft** [reply]
Uhhh you where quick. I was quick too, there is a pull request with the keep option made by myself already.

Would be very nice if you accept this pull request BEFORE throwing Laravel 10 and 11 over board.

**Reply by stefanzweifel** [reply]
@ojsoft Sorry. Only now seen your message. 🤦 
Unfortunately the `--keep` option is now bundled in the same release as dropping support for older PHP and Laravel versions.

**Comment by stefanzweifel** [comment]
Closing this issue as it’s out of scope for this package.

I don’t use this package to restore files and don’t see a need for that functionality in the future. As mentioned in earlier comments, this is a hard problem to solve, and not one I have the time or energy to take on.

Going into 2026 and beyond, I’m intentionally limiting new development to features I actively need or that clearly benefit the package as a whole.

If this functionality is important to you, feel free to fork the package and implement it yourself. A well-thought-out pull request is welcome, and I’m happy to review it when time allows.
